tests/pytests/test_shard_window_ratio.py

```diff
@@ -0,0 +1,439 @@
+    conn.execute_command('HSET', 'doc1', 'v', vector.tobytes())
+    
+    waitForIndex(env, 'idx')
+
+    query_vec = create_np_array_typed([1] * dim, 'FLOAT32')
+
+    # Test invalid ratio values
+    invalid_ratios = [0.0, -0.1, 1.1, 2.0, "invalid"]
+    
+    for ratio in invalid_ratios:
+        # Should return error for invalid ratios
+        env.expect('FT.SEARCH', 'idx',
+                  f'*=>[KNN 10 @v $query_vec]=>{{$shard_window_ratio: {ratio}}}',
+                  'PARAMS', 2, 'query_vec', query_vec.tobytes()).error()
+    
+    # Test malformed syntax scenarios
+    env.debugPrint("\n=== Testing malformed syntax scenarios ===", force=True)
+    
+    malformed_queries = [
+        # Missing closing brace
+        '*=>[KNN 10 @v $query_vec]=>{$shard_window_ratio: 0.5',
+        # Missing colon
+        '*=>[KNN 10 @v $query_vec]=>{$shard_window_ratio 0.5}',
+    ]
+    
+    for malformed_query in malformed_queries:
+        env.expect('FT.SEARCH', 'idx', malformed_query,
+                  'PARAMS', 2, 'query_vec', query_vec.tobytes()).error()
+
+
+    # Test invalid ratio values for FT.AGGREGATE
+    for ratio in invalid_ratios:
+        # Should return error for invalid ratios
+        env.expect('FT.AGGREGATE', 'idx',
+                  f'*=>[KNN 10 @v $query_vec]=>{{$shard_window_ratio: {ratio}}}',
+                  'PARAMS', 2, 'query_vec', query_vec.tobytes()).error()
+
+    # Test malformed syntax scenarios for FT.AGGREGATE
+    for i, malformed_query in enumerate(malformed_queries):
+        env.debugPrint(f"Testing malformed query {i}: {malformed_query}", force=True)
+        env.expect('FT.AGGREGATE', 'idx', malformed_query,
+                  'PARAMS', 2, 'query_vec', query_vec.tobytes()).error()
+
+
+
+def test_ft_profile_shard_result_validation_scenarios():
+    """Test comprehensive scenarios for shard window ratio validation."""
+    env = Env(moduleArgs='DEFAULT_DIALECT 2')
+    conn = getConnectionByEnv(env)
+
+    # Create index
+    dim = 64
+    env.expect('FT.CREATE', 'idx_scenarios', 'SCHEMA',
+               'v', 'VECTOR', 'HNSW', '6', 'TYPE', 'FLOAT32', 'DIM', dim, 'DISTANCE_METRIC', 'L2').ok()
+
+    # Add test documents - ensure we have enough for all test scenarios
+    num_docs = 2000  # Increased to ensure we always have enough results
+    p = conn.pipeline(transaction=False)
+    for i in range(1, num_docs + 1):
+        vector = create_np_array_typed([i % 100] * dim, 'FLOAT32')
+        p.execute_command('HSET', f'doc{i}', 'v', vector.tobytes())
+    p.execute()
+
+    waitForIndex(env, 'idx_scenarios')
+
+    query_vec = create_np_array_typed([50] * dim, 'FLOAT32')
+
+    # Test scenarios with different characteristics
+    # Expected results depend on cluster vs standalone mode
+    scenarios = [
+        {
+            'k': 3,
+            'ratio': 0.3,
+            'expected_effective_k': 1,
+            'expected_total_results_cluster': 3,  # User always gets K results in cluster
```

Comment:
probably will have to set it later as it depends on a previous pentry of this dict
```suggestion
            'expected_total_results_cluster': expected_effective_k * num_shards,  # User always gets K results in cluster
```